Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Repository.rst

```diff
@@ -65,6 +71,33 @@ several custom find methods, two of them are shown below:
 
 ..  include:: /CodeSnippets/Extbase/Domain/CustomMethods.rst.txt
 
+
+Magic find methods
+==================
+
+..  deprecated:: 12.3
+    As these methods are widely used in almost all Extbase-based extensions,
+    they are marked as deprecated in TYPO3 v12, but will only trigger a
+    deprecation notice in TYPO3 v13, as they will be removed in TYPO3 v14.
+    Migrate the usage of these methods to the new
+    :ref:`find methods <extbase-repository-find-methods>`.
+
+The :php:`Repository` class creates "magic" methods to find by attributes of
+model.
+
+:php:`findBy[Property]`
+   Finds all objects with the provided property.
+
+:php:`findOneBy[Property]`
+   Returns the first object found with the provided property.
+
+:php:`countBy[Property]`
+   Counts all objects with the provided property.
+
+If necessary, these methods can also be overridden by implementing them in the
+concrete repository.
+
+
```

Comment:
I would love to see some examples on how to migrate from the magic find methods with examples here.